addons/network/XEH_preInit.sqf

```diff
@@ -0,0 +1,42 @@
+#include "script_component.hpp"
+
+ADDON = false;
+
+// Restore loadouts lost by the naked unit bug
+[QGVAR(restoreLoadout), {
+    params ["_unit", "_loadout"];
+    if (count _loadout < 7) exitWith {};
+
+    // Check if loadout got lost
+    private _loadoutLost = False;
+    {
+        (_loadout select _forEachIndex) params [["_backup", "", [""]]];
+        // Reject backup if it contains less information
+        if (_backup isEqualTo "" && !(_x isEqualTo "")) exitWith {};
+        if !(_x isEqualTo _backup) exitWith {
+            _loadoutLost = True;
+        };
+    } forEach [
+        primaryWeapon _unit,
+        secondaryWeapon _unit,
+        handgunWeapon _unit,
+        uniform _unit,
+        vest _unit,
+        backpack _unit,
+        headgear _unit
+    ];
```

Comment:
If the unit is supposed to be naked, why check all this stuff?

Reply:
They aren't guaranteed to be naked that's the bug, it doesn't always happen. Most notable cases are loss of everything, but sometimes they lose only vest, or weapon etc.

We should avoid setting loadout if unnecessary I think.